### DISCUSSION TITLE
React

### DISCUSSION THREAD

**Discussion by DesmondDallas** [vercel/next.js#56722] [Show and tell] [answered]
Hi, I have returned back to React website. It has changed and seems to push me towards Next.js which is new to me. Am I missing something and this is the new way forward for React?

**Comment by icyJoseph** [comment]
Well, to make production grade apps, Next.js is quite good, the Pages Router has been a solid option for quite a while, and the App Router is gaining traction.

But if you want to first get a good React learning experience, just type `react.new` in your browser URL search bar, and that'll bring up a codesandbox ready to go, you can also go to `next.new` and that'll bring a Stackblitz Next.js project ready to run, both in your browser.

**Reply by DesmondDallas** [reply]
Thank you, much appreciated. This maybe a novice questioning but I would like to clarify what exactly you mean by production grade app. I understand it as an app I can build ready to put on the server for users. Doesn't this stand for React also, what is the difference in simple terms please?

**Reply by DesmondDallas** [reply]
Hi Joseph, thought I would take the opportunity to ask. I've been learning web development on and of for quite a while. HTML, CSS, JavaScript. I think I need to do a structured course to grasp the foundation of programming, as I'm feeling as though I'm all over the place. Do you recommend any cost efficient reputable courses. My work uses c#, .net which would honestly be the most sensible language to learn for me, in hope to get a foot into the tech department.

**Reply by icyJoseph** [reply]
Hi,

You are right of course with Vite or CRA you could also make an app, that you can ship. What I meant was more about:

- Ability to leverage a server
- Ability to serve entities that do not execute JS
- Control over bundling
- Many developers are involved
- You have tooling/APIs that helps you focus on business code
- Keep other tools under control, such as Webpack, etc

Long ago I gave a presentation: https://windy-fold.surge.sh/#/normal/0/-1 on Next.js ~ 

Regarding courses, it depends, if you first want to grasp React, basics, patterns, advanced hooks, performance etc, there's Josh Comeau's "Joy of React", and react.gg courses, which are very nice. Josh's even ends with App Router. Epic React by Kent C Dodds is also good, and the course material is Open Sourced, but the certification with videos does cost money.

To put it in perspective, if you master React and the immediate surrounding to the React state management tools, as well as the job React does, which is rendering HTML, you'd have gained a lot of ground. However, there's more parts to a development stack, doing work server side, optimising various metrics, SEO, styling solutions, etc... that's where something like Next.js or Remix kicks in, to help you gain even more ground.

**Comment by DesmondDallas** [comment]
Thank you,

I will take a look. Your feedback is very helpful and much appreciated.

**Reply by icyJoseph** [reply, marked as answer]
No worries, don't forget to mark as answered, or closed, and feel free to keep on asking :)
